shared/src/commonMain/kotlin/org/hyperskill/app/step_completion/presentation/StepCompletionActionDispatcher.kt

```diff
@@ -111,19 +115,31 @@ class StepCompletionActionDispatcher(
                 if (topicProgress.isCompleted) {
                     topicCompletedFlow.notifyDataChanged(action.topicId)
 
-                    val topicTitle = topicsInteractor
-                        .getTopic(action.topicId)
-                        .map { it.title }
-                        .getOrElse { return onNewMessage(Message.CheckTopicCompletionStatus.Error) }
+                    coroutineScope {
+                        val topicTitleResult = async {
+                            topicsInteractor.getTopic(action.topicId)
+                        }
+                        val nextStepIdResult = async {
+                            topicsToDiscoverNextInteractor.getNextTopicToDiscover()
+                        }
+
+                        val topicTitle = topicTitleResult.await()
+                            .map { it.title }
+                            .getOrElse { return@getOrElse onNewMessage(Message.CheckTopicCompletionStatus.Error) }
```

Comment:
`return@getOrElse` should be replaced with `return@coroutineScope`, to complete execution of coroutineScope block.
Otherwise the type of topicTitle would be `Any`, instead of `String`, and execution of the block will be continued.